lib/pure/cstrutils.nim

```diff
@@ -12,20 +12,22 @@
 ## save allocations.
 
 include "system/inclrtl"
+import std/private/strimpl
 
-proc toLowerAscii(c: char): char {.inline.} =
-  if c in {'A'..'Z'}:
-    result = chr(ord(c) + (ord('a') - ord('A')))
-  else:
-    result = c
 
 when defined(js):
   proc startsWith*(s, prefix: cstring): bool {.noSideEffect,
     importjs: "#.startsWith(#)".}
 
   proc endsWith*(s, suffix: cstring): bool {.noSideEffect,
     importjs: "#.endsWith(#)".}
-  
+
+  proc cmpIgnoreStyle*(a, b: cstring): int {.noSideEffect.} =
+    cmpIgnoreStyleImpl(a, b)
+
+  proc cmpIgnoreCase*(a, b: cstring): int {.noSideEffect.} =
```

Comment:
`func` ?.

Reply:
Not the scope of this PR, as I said before the whole module will be cleaned in the following PR.
https://github.com/nim-lang/Nim/pull/16399#issue-542862613

```
proc + noSideEffect => func
double backticks => single backtick
```